### DISCUSSION TITLE
Pointer to gaussian.c algorithm literature?

### DISCUSSION THREAD

**Discussion by Hermann-SW** [alpertron/calculators#20] [Q&A] [answered]
I am a big fan of your gaussian.c -- I use it for determining unique sum of two squares for primes ≡1 (mod 4).
I play a lot with RSA numbers:
https://gist.github.com/Hermann-SW/839dfe6002810d404e3f0fe1808a6333
Running gaussian for 120-digit prime factor of already factored RSA-240 took 0.1s on Raspberry Pi400.
Computation took 3s for 2082-digit prime on Intel i7.
And even for 10001-digit prime number only 2:32min.
That is the reason I forked this repo, used only gaussian and made Python verifyable output:
```
./square_sum_prod 97
(9**2 + 4**2) == 97

./square_sum_prod 693342667110830181197325401899700641361965863127336680673013|\
python -c 'import sys;l=sys.stdin.read().rstrip();print(l,"\n"+str(eval(l)))'
(832050675380196295918334279642**2 + 32161167736719180733109944457**2) == 693342667110830181197325401899700641361965863127336680673013 
True
```

Do you have any literature pointers for algorithm used for gaussian.c?
Here fork mission statement:
https://github.com/Hermann-SW/square_sum_prod#this-fork-mission-statement

**Comment by alpertron** [comment, marked as answer]
Hermann,

Thanks for using my program.

You can see the methods I used at: https://www.alpertron.com.ar/4SQUARES.HTM

There are more optimizations in the calculator but I had no time to add them to the documentation.

Best regards,
Darío Alpern

**Reply by Hermann-SW** [reply]
THANKS!

Exactly what I looked for.
And I need all, primes ≡1 (mod 4), ≡3 (mod 4) and non-primes.

Today I made square_sum_prod() available in RSA_numbers_factored.py repo:
https://gist.github.com/Hermann-SW/839dfe6002810d404e3f0fe1808a6333
https://forums.raspberrypi.com/viewtopic.php?p=2059748#p2059748

I did it quick and dirty via subprocess.Popen(), knowing that dirty will remain long after qick is gone.
But since square_sum_prod forked from your repo is superfast, that is not really a problem.
This demo determines biggest first prime factor of RSA numbers factored sofar, outputs its size (digits+bits), determines unique sum of squares, pretty prints and verifies the identity:
<img src="https://forums.raspberrypi.com/download/file.php?id=57933">
